### DISCUSSION TITLE
[提问] 有关mod开发的一些疑问

### DISCUSSION THREAD

**Discussion by kurt-steiner** [mouse0w0/MinecraftDeveloperGuide#23] [通用 / General]
第一次接触这种游戏插件开发，想问一下文档里那么多插件种类，他们有什么特点，我该选哪个

**Comment by SettingDust** [comment]
**选择依赖于你选择的平台**

数据包是原版机制，可以通过编写符合标准的数据包来达成有限的编辑，较为简易，效果也有限
Forge，Fabric 是 Mod Loader 及 Mod 开发框架，即向游戏（服务端、客户端）加载模组、提供模组 API
Bukkit，Sponge 是服务端，即通过实现 MC 的通信协议来承载多人游戏，二者都有相应的插件 API
BungeeCord，Velocity 是代理，通过对网络数据的转发来实现跨服（程序意义上的，具体同步、传送之类的功能不是它们本身就有的），它们也有相应 API

需要注意的是，这些项目都有多个部分，实现不同的功能，上述只是回答你的问题，即它们的特点，且上述的项目都有不同的分支版本，如：
Sponge：Lantern（实现、API 不同）
Bukkit：Spigot，PaperSpigot（优化）
BungeeCord：Waterfall（兼容 Forge）

综上所述，你希望添加怪物、物品，那只能选择模组开发，你希望开好一个服务器，那就选择一个服务端来开发，你想开一个群组服来承载更多的人，那就需要学习代理的插件开发。

即你现在想这个太早了，需要用自然就知道了:D

**Comment by sileence114** [comment]
比起开发，你应该先学会安装这些服务端、安装插件。

慢慢地，你就能了解到这些插件的局限性，或者某些API的局限性。这些“局限性”就是这些平台/API的特点。

**Comment by Hyffer** [comment]
没太明白你想问的是mod还是插件(plugin)，但是一定注意这两者是不同的东西（上面那位SettingDust的分类十分详细）。我没接触过mod开发，但就我写插件的经验来看，插件更适合多人游戏玩法的拓展，而mod可能更倾向于更改世界的生成、添加物品及生物（典型的例子像是起床战争、空岛生存插件，虚无世界、工业2模组）

定下大体的方向，选定插件或是mod，接下来遇到什么问题解决什么问题就好了。

对于新手的话，我觉得少说一些更好xD

**Comment by luqin2007** [comment]
那么多版本如果分不清的话，你需要回答以下几个问题
- 我的平台是什么？确定服务器类型或mod加载平台，最起码要确定下是插件还是mod。
- 我要做什么？能不能用原版数据包或者其他mod实现，比如ct等。很多为了服务器或整合包需要实现的功能不需要单独写一个mod
- 版本呢？mc各个版本差距还是挺大的
